core/resource/resource.go

```diff
@@ -94,6 +94,10 @@ func (r *Resource) UpdateURN(urn string) error {
 	return errors.InvalidArgument(EntityResource, "urn already present for "+r.FullName())
 }
 
+func (r *Resource) UpdateTenant(tnnt tenant.Tenant) {
+	r.tenant = tnnt
```

Comment:
Please add some validation in here. An example, whether we will allow update in project or we just restrict to namespace only.

Reply:
for now, this is supposed to change the complete tenant

Reply:
sure